### PR TITLE
[Form] turn failed file uploads into form errors

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #27524
| License       | MIT
| Doc PR        | 

#FOSSHackathons #EUFOSSA

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/FileType.php [hunk 1]
```diff
@@ -22,6 +23,15 @@
 
 class FileType extends AbstractType
 {
+    const KIB_BYTES = 1024;
+    const MIB_BYTES = 1048576;
+
+    private static $suffices = [
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Core/Type/FileType.php]: suffixes

File: src/Symfony/Component/Form/Extension/Core/Type/FileType.php [hunk 2]
```diff
@@ -116,4 +132,96 @@ public function getBlockPrefix()
...
+            case 'm': $max *= 1024;
+            // no break
+            case 'k': $max *= 1024;
+        }
+
+        return $max;
+    }
+
+    /**
+     * Convert the limit to the smallest possible number
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Core/Type/FileType.php]: Converts

**Review by HeahDude — APPROVED** [review]
LGTM

**Comment by fabpot** [comment]
@xabbuh Are the test failures related?

**Comment by xabbuh** [comment]
failures should have been fixed in the meantime by #30905

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**